sap_business_document_processing/document_information_extraction_client/dox_api_client.py

```diff
@@ -0,0 +1,608 @@
+        comma separated string. If none are given, all available header fields are used
+        :param line_item_fields: (optional) A list of line item fields to be extracted. Can be passed as list of string
+        or as comma separated string. If none are given, all available line item are used
+        :param template_id: (optional) The ID of the template to be used for this document
+        :param received_date: (optional) The date the document was received
+        :param enrichment: (optional) A dictionary of entities that should be used for entity matching
+        :param return_null_values: Flag if fields with null as value should be included in the response or not.
+        Default is False
+        :param wait_for_results: Flag if the method will wait for the processed results. Default is True.
+        :param silent: If True, the functions returns even if some documents failed uploading or processing. If False,
+        will raise an exception if at least one document failed. Default is False
+        :return: An iterator with results for successful documents and exceptions for failed documents. Use
+        next(iterator) within a try-catch block to filter the failed documents. If input argument `wait_for_results` is
+        true, the iterator will contain processed document results. If false, the iterator will contain information
+        about the uploaded document including its ID; the user can use `get_document_results` method to further fetch
+        the processed results given the iterator.
+        """
+        if not isinstance(document_paths, list) or len(document_paths) == 0:
+            raise ValueError(f'Argument \'document_paths\' must be a list of paths to documents to be uploaded, '
+                             f'but got {document_paths}')
+        number_of_documents = len(document_paths)
+
+        options = merge_document_options(options, client_id, document_type, header_fields, line_item_fields,
+                                         template_id, received_date, enrichment)
+
+        self.logger.debug(f'Starting upload of {number_of_documents} documents for client {client_id} in parallel using '
+                          f'{self.polling_threads} threads')
+        with ThreadPoolExecutor(min(self.polling_threads, number_of_documents)) as pool:
+            upload_results = pool.map(self._single_upload_wrap_errors, document_paths, [options] * number_of_documents)
+        if not silent:
+            upload_results = self._validate_results(upload_results, 'Some documents could not be uploaded successfully')
+
+        self.logger.info(f'Finished uploading {number_of_documents} documents for client {client_id}')
+        if not wait_for_results:
+            return self._create_result_iterator(upload_results)
+
+        return self.get_documents_results(upload_results, return_null_values, silent)
+
+    def get_documents_results(self, upload_results, return_null_values=False, silent=False):
```

Comment:
In which use-case would user need this as a public method?

Reply:
When a user wants to upload document with upload_documents and sets the parameter wait_for_results wrong, but later wants to get all of those processed documents, they can provide this function with the result of the previous called function, to get all results.

Reply:
I see, this is related to the next comment, let's see how that discussion resolves and check later on whether this remark is still relevant.